code/Test_definitions/carrier-billing-refund-retrievePaymentRemainingAmount.feature

```diff
@@ -0,0 +1,204 @@
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+
+  @retrieve_payment_remaining_amount_401.02_expired_access_token
+  Scenario: Expired access token
+    Given the header "Authorization" is set to an expired access token
+    And the path parameter "paymentId" is set to a valid value
+    When the HTTP "GET" request is sent
+    Then the response status code is 401
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+
+  @retrieve_payment_remaining_amount_401.03_invalid_access_token
+  Scenario: Invalid access token
+    Given the header "Authorization" is set to an invalid access token
+    And the path parameter "paymentId" is set to a valid value
+    When the HTTP "GET" request is sent
+    Then the response status code is 401
+    And the response header "Content-Type" is "application/json"
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+
+  # Error 403 scenarios
+
+  @retrieve_payment_remaining_amount_403.01_invalid_token_permissions
+  Scenario: Inconsistent access token permissions
+    # To test this, an access token has to be obtained without carrier-billing-refund:refunds:read scope
+    Given header "Authorization" is set to a valid access token emitted without carrier-billing-refund:refunds:read scope
+    And the path parameter "paymentId" is set to a valid value
+    When the HTTP "GET" request is sent
+    Then the response status code is 403
+    And the response property "$.status" is 403
+    And the response property "$.code" is "PERMISSION_DENIED"
+    And the response property "$.message" contains a user friendly text
+
+
+  @retrieve_payment_remaining_amount_403.02_phoneNumber_token_mismatch
+  Scenario: Inconsistent access token context for the phoneNumber
+    # To test this, a 3-legged access token has to be obtained for a different phoneNumber
+    Given the header "Authorization" is set to a valid access token emitted for a phone number
+    And the path parameter "paymentId" is set to a valid value associated to a phone number different than the access token is associated
+    When the HTTP "GET" request is sent
+    Then the response status code is 403
+    And the response property "$.status" is 403
+    And the response property "$.code" is "INVALID_TOKEN_CONTEXT"
+    And the response property "$.message" contains a user friendly text
+
+
+  # Error 404 scenarios
+
+  @retrieve_payment_remaining_amount_404.01_payment_not_found
+  Scenario: Payment not found
+    # To test this, a 2-legged access token is needed, just beacuse if not it triggers test CHECK "@retrieve_payment_remaining_amount_403.02_phoneNumber_token_mismatch"
```

Comment:
I'm not getting why 2-legs access is required for this scenario. Because if we are in 3legs situation the scenario @retrieve_payment_remaining_amount_403.02_phoneNumber_token_mismatch covers it?

because instead of beacuse

Reply:
Yes, you are right @bigludo7.

In 2-legged case, paymentId has to be related to API ClientId
In 3-legged case, paymentId haas to be realted to API ClientId+User PhoneNumber

Otherwise, 404 Not Found


Removing the comment